examples/generative/cyclegan.py

```diff
@@ -0,0 +1,695 @@
+    if activation:
+        x = activation(x)
+    return x
+
+
+def upsample(
+    x,
+    filters,
+    activation,
+    kernel_size=(3, 3),
+    strides=(2, 2),
+    padding="same",
+    kernel_initializer=kernel_init,
+    gamma_initializer=gamma_init,
+    use_bias=False,
+):
+    x = layers.Conv2DTranspose(
+        filters,
+        kernel_size,
+        strides=strides,
+        padding=padding,
+        kernel_initializer=kernel_initializer,
+        use_bias=use_bias,
+    )(x)
+    x = tfa.layers.InstanceNormalization(gamma_initializer=gamma_initializer)(x)
+    if activation:
+        x = activation(x)
+    return x
+
+
+"""
+## Build the generators
+
+The generator consists of downsampling blocks, two residual blocks
+and upsampling blocks. The structure of the generator is like this:
+
+c7s1-64 ==> Conv block with `relu` activation, filter size of 7
+d128 ====|
+         |-> 2 downsampling blocks
+d256 ====|
+R256 ====|
+R256     |
+R256     |
+R256     |
+R256     |-> 9 residual blocks
+R256     |
+R256     |
+R256     |
+R256 ====|
+u128 ====|
+         |-> 2 upsampling blocks
+u64  ====|
+c7s1-3 => Last conv block with `tanh` activation, filter size of 1.
+"""
+
+
+def get_resnet_generator(
+    filters=64,
+    num_downsampling_blocks=2,
+    num_residual_blocks=9,
+    num_upsample_blocks=2,
+    gamma_initializer=gamma_init,
+    name=None,
+):
+    img_input = layers.Input(shape=input_img_size, name=name + "_img_input")
+
+    x = ReflectionPadding2D(padding=(3, 3))(img_input)
+    x = layers.Conv2D(filters, (7, 7), kernel_initializer=kernel_init, use_bias=False)(
+        x
+    )
+    x = tfa.layers.InstanceNormalization(gamma_initializer=gamma_initializer)(x)
+    x = layers.Activation("relu")(x)
+
+    # Downsampling
+    for _ in range(num_downsampling_blocks):
+        filters *= 2
+        x = downsample(x, filters=filters, activation=layers.Activation("relu"))
+
+    # Residual blocks
+    for _ in range(num_residual_blocks):
+        x = residual_block(x, activation=layers.Activation("relu"))
+
+    # Upsampling
+    for _ in range(num_upsample_blocks):
+        filters //= 2
+        x = upsample(x, filters, activation=layers.Activation("relu"))
+
+    # Final block
+    x = ReflectionPadding2D(padding=(3, 3))(x)
+    x = layers.Conv2D(3, (7, 7), padding="valid")(x)
+    x = layers.Activation("tanh")(x)
+
+    model = keras.models.Model(img_input, x, name=name)
+    return model
+
+
+"""
+## Build the discriminators
+
+The discriminators have the following architecture:
+`C64->C128->C256->C512`
+"""
+
+
+def get_discriminator(
+    filters=64, kernel_initializer=kernel_init, num_downsampling=3, name=None
+):
+    img_input = layers.Input(shape=input_img_size, name=name + "_img_input")
+    x = layers.Conv2D(
+        filters,
+        (4, 4),
+        strides=(2, 2),
+        padding="same",
+        kernel_initializer=kernel_initializer,
+    )(img_input)
+    x = layers.LeakyReLU(0.2)(x)
+
+    x = downsample(
```

Comment:
You could simply this model definition by using a `for` loop iterating over parameters, calling `downsample`